framework/include/controls/ControlInterface.h

```diff
@@ -129,98 +86,59 @@ class ControlInterface
   const std::string & _ci_name;
 
   // Only these objects are allowed to construct this interface
-  friend class RealParameterReporter;
+  friend class RealControlParameterReporter;
   friend class Control;
 };
 
 template<typename T>
-T &
+ControllableParameter<T>
 ControlInterface::getControlParam(const std::string & name)
 {
-  return getControlParamByName<T>(_ci_parameters.get<std::string>(name));
+  return getControlParamByName<T>(_ci_parameters.get<std::string>(name),
+                                  std::numeric_limits<unsigned int>::max());
 }
 
 template<typename T>
-T &
-ControlInterface::getControlParamByName(const std::string & name)
+ControllableParameter<T>
+ControlInterface::getControlParam(const std::string & name, unsigned int count)
```

Comment:
I just don't think that's a good idea.  You might have made the `Control` thinking of only controlling one thing... but now someone wants to use it to control more.  Why hardcode a limitation into the class itself?